test/integration/Integration.ts

```diff
@@ -278,162 +326,198 @@ describe('Integration', function () {
+
+            const attestationToEpochKey = new Attestation(
+                BigInt(attesters[0].id),
+                BigInt(0),
+                BigInt(DEFAULT_POST_KARMA),
+                BigInt(0),
+                false,
+            )
+            
+            const tx = await unirepContractCalledByFirstAttester.publishPost(
+                BigInt(add0x(postId)), 
+                epochKey,
+                postText, 
+                publicSignals, 
+                proof,
+                nullifiers,
+                { value: attestingFee, gasLimit: 1000000 }
             )
+
             const receipt = await tx.wait()
-            expect(receipt.status, 'Submit attestation failed').to.equal(1)
+            expect(receipt.status, 'Submit post failed').to.equal(1)
 
             secondEpochEpochKeys.push(firstUserEpochKey.toString())
-            unirepState.addAttestation(firstUserEpochKey.toString(), attestation)
+            unirepState.addAttestation(firstUserEpochKey.toString(), attestationToEpochKey)
+            for(const user of users){
+                user.updateAttestation(firstUserEpochKey, attestationToEpochKey.posRep, attestationToEpochKey.negRep)
+            }
         })
 
-        it('Second attester attest to first user', async () => {
+        it('Second attester upvote to first user', async () => {
             const nonce = 0
             const firstUserEpochKey = genEpochKey(users[0].id.identityNullifier, currentEpoch.toNumber(), nonce, circuitEpochTreeDepth)
             const graffitiPreImage = genRandomSalt()
-            const attestation: Attestation = new Attestation(
-                attesters[1].id,
-                BigInt(3),
-                BigInt(1),
-                hashOne(graffitiPreImage),
-                true,
-            )
+            const secondUserEpochKey = genEpochKey(users[1].id.identityNullifier, currentEpoch.toNumber(), nonce, circuitEpochTreeDepth)
+
+            // gen nullifier nonce list
+            const voteValue = 3
+            const proveKarmaAmount = voteValue
+            const upvoteValue = BigInt(voteValue)       
+            const nonceStarter: number = 0
+   
+            // gen minRep proof
+            const minRep =  0
+
             // Add graffiti pre-image to graffitiPreImageMap
-            graffitiPreImageMap[0][attestation.attesterId.toString()] = graffitiPreImage
-            console.log(`Attester attest to epk ${firstUserEpochKey} with ${attestation.toJSON()}`)
-            const tx = await unirepContractCalledBySecondAttester.submitAttestation(
-                attestation,
-                firstUserEpochKey,
-                { value: attestingFee }
+            graffitiPreImageMap[0] = new Object()
+            graffitiPreImageMap[0][attesters[1].id] = graffitiPreImage
+            console.log(`Attester attest to epk ${firstUserEpochKey} with vote value ${voteValue}`)
```

Comment:
Is this debug message still valid?

Reply:
Yes it executes the same logic as the message.
Maybe it is better to replace `attest` by `upvote`.